app/services/bibid_updater.rb

```diff
@@ -0,0 +1,78 @@
+# frozen_string_literal: true
+
+# Service to update Voyager bibids to alma mms ids
+class BibidUpdater
+  def self.update(logger: Logger.new(STDOUT), batch_size: 1000)
+    new(
+      query_service: Valkyrie::MetadataAdapter.find(:postgres).query_service,
+      logger: logger,
+      batch_size: batch_size
+    ).update
+  end
+
+  attr_reader :query_service, :logger, :batch_size
+  delegate :run_query, to: :query_service
+  def initialize(query_service:, logger:, batch_size: 1000)
+    @query_service = query_service
+    @logger = logger
+    @batch_size = batch_size
+  end
+
+  def update
+    logger.info "Updating #{total_count} records"
+    progress_bar
+    (1..iterations).each do
+      run_query(update_query)
+      progress_bar.progress += 1
+    end
+
+    Reindexer.reindex_all
+  end
+
+  private
+
+    def iterations
+      @iterations ||= 1 + (total_count / batch_size)
+    end
+
+    def progress_bar
+      @progress_bar ||= ProgressBar.create format: "%a %e %P% Batches Processed: %c of %C", total: iterations
+    end
+
+    def total_count
+      @total_count ||= query_service.connection[total_count_query].first[:count]
+    end
+
+    def total_count_query
+      <<-SQL
+        SELECT COUNT(*)
+        #{voyager_resources_query}
+      SQL
+    end
+
+    def update_query
```

Comment:
issue: A concern I have about this is it won't update the metadata in preservation storage or re-fetch metadata from the alma records to make sure it's okay.